File: projects/number-guessing/java/Guess.java
import java.util.Random;
import java.util.Scanner;

public class Guess {
    public static void main(String[] args) {
        Random rand = new Random();
        int secret = rand.nextInt(100) + 1;
        int attempts = 0;
        Scanner sc = new Scanner(System.in);

        System.out.println("Guess the number (1-100)");

        while (true) {
            System.out.print("Your guess: ");
            int guess = sc.nextInt();
            attempts++;

            if (guess < secret) System.out.println("Too low");
            else if (guess > secret) System.out.println("Too high");
            else {
                System.out.println("Correct! Attempts: " + attempts);
                break;
            }
        }
        sc.close();
    }
}

File: projects/number-guessing/javascript/guess.js
const readline = require("readline").createInterface({
  input: process.stdin,
  output: process.stdout
});

const secret = Math.floor(Math.random() * 100) + 1;
let attempts = 0;

console.log("Guess the number (1-100)");

function ask() {
  readline.question("Your guess: ", answer => {
    const guess = Number(answer);
    attempts++;

    if (guess < secret) {
      console.log("Too low");
      ask();
    } else if (guess > secret) {
      console.log("Too high");
      ask();
    } else {
      console.log(`Correct! Attempts: ${attempts}`);
      readline.close();
    }
  });
}

ask();
